content/advent-2015/etcd-distributed-key-value-store-with-grpc-http2.md

```diff
@@ -0,0 +1,375 @@
++++
+author = ["Gyu-Ho Lee"]
+date = "2015-12-29T08:00:00+00:00"
+linktitle = "etcd: distributed key-value store with grpc/http2"
+series = ["Advent 2015"]
+title = "etcd: distributed key-value store with grpc/http2"
++++
+
+## What is etcd?
+
+<img alt="etcd"
+     src="https://raw.githubusercontent.com/coreos/etcd/master/logos/etcd-horizontal-color.png"
+     style="float:right; margin-bottom:2em;"/>
+
+etcd is a distributed, consistent key-value store, written in Go. Similar to how Linux distributions typically use `/etc` to store local configuration data, etcd can be thought of as a reliable store for *distributed* configuration data. It is distributed by replicating data to multiple machines, therefore highly available against single point of failures. Using Raft consensus algorithms, etcd gracefully handles network partitions and machine failures, even leader failures. etcd is being widely used in production: CockroachDB, CoreOS, Kubernetes, etc.
```

Comment:
nit: cockroachdb doesn't use etcd directly but we share a raft library: https://godoc.org/github.com/coreos/etcd/raft